# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestBitWriter.cpp
# Commit: 412f97da1656
# Full Hash: 412f97da16567da3a0c2a6adc6dbde27a28dea22
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-11 09:22:03
# Regressor Bug: 1852333
# File Overlap Count: 4
# Description:
#   Bug 1852333 - part2 : detect config change for HEVC stream. r=media-playback-reviewers,padenot
#   
#   This patch detects inband change for the HEVC playback so that we can
#   recreate decoder if the new config is different from the previous one.
#   
# ==============================================================================

diff -r b15b2e7c785a -r 412f97da1656 dom/media/gtest/TestBitWriter.cpp
--- a/dom/media/gtest/TestBitWriter.cpp	Wed Oct 11 00:57:39 2023 +0000
+++ b/dom/media/gtest/TestBitWriter.cpp	Wed Oct 11 00:57:40 2023 +0000
@@ -3,6 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include <stdint.h>
 #include "gtest/gtest.h"
 #include "BitReader.h"
 #include "BitWriter.h"
@@ -56,6 +57,25 @@
   EXPECT_EQ(length, BitReader::GetBitLength(test));
 }
 
+TEST(BitWriter, AdvanceBytes)
+{
+  RefPtr<MediaByteBuffer> test = new MediaByteBuffer();
+  BitWriter b(test);
+  b.WriteBits(0xff, 8);
+  EXPECT_EQ(test->Length(), 1u);
+
+  uint8_t data[] = {0xfe, 0xfd};
+  test->AppendElements(data, sizeof(data));
+  EXPECT_EQ(test->Length(), 3u);
+  b.AdvanceBytes(2);
+
+  b.WriteBits(0xfc, 8);
+  EXPECT_EQ(test->Length(), 4u);
+
+  BitReader c(test);
+  EXPECT_EQ(c.ReadU32(), 0xfffefdfc);
+}
+
 TEST(BitWriter, SPS)
 {
   uint8_t sps_pps[] = {0x01, 0x4d, 0x40, 0x0c, 0xff, 0xe1, 0x00, 0x1b, 0x67,